### DISCUSSION TITLE
AutoGluon Textpredicter (Confusion Matrix, ROC, AUC)

### DISCUSSION THREAD

**Discussion by Recep1997** [autogluon/autogluon#2037] [Q&A]
Hello, 

I am currently working on a project involving text classification with AutoGluon Textpredicter. For the evaluation I would like to set up a confusion matrix as well as visualize a ROC. Unfortunately I can't get it to work that way. The function for the confusion matrix (Predictor.evaluate_predictions()) doesn't work somehow and I don't know what to do. Can someone give me a tip or has an example for me. I work with the Jupiter Notebook from Anaconda. A .ipynb file with an example would be very helpful for me.

With kind regards

**Comment by sxjscience** [comment]
@Recep1997 You can use `predict_proba` https://github.com/awslabs/autogluon/blob/6b8c23d9c62b39940e45ed8fc395116f3f8d3587/multimodal/src/autogluon/multimodal/predictor.py#L1770 and calculate the confusion matrix with the probabilities.
